frontend/src/metabase/dashboard/components/DashboardHeader/buttons/NightModeToggleButton.tsx

```diff
@@ -9,15 +9,17 @@ export const NightModeToggleButton = ({
   onNightModeChange,
 }: {
   isNightMode: boolean | undefined;
-  onNightModeChange: (isNightMode: boolean) => void;
+  onNightModeChange?: (isNightMode: boolean) => void;
 }) => (
-  <Tooltip tooltip={isNightMode ? t`Daytime mode` : t`Nighttime mode`}>
+  <Tooltip
+    tooltip={isNightMode ? t`Daytime mode` : t`Nighttime mode`}
+  >
     <span>
       <DashboardHeaderButton
         icon={
           <NightModeButtonIcon
             isNightMode={isNightMode}
-            onClick={() => onNightModeChange(!isNightMode)}
+            onClick={() => onNightModeChange?.(!isNightMode)}
```

Comment:
Can you double-check the type if this could actually be `undefined`? At least, from this component's point of view, `onNightModeChange` is always passed.

Reply:
Ok I had a look and it's actually the SDK instances (`InteractiveDashboard` and `StaticDashboard`) where we don't pass the night mode controls

Reply:
I'm going to put `false` and `noop` for the SDK instances since I think it makes more sense to pass the embed display values down the chain without the possibility of undefined. It'll make our code a slightly nicer I think.

Reply:
Then can you ensure you revert this line change (removing the optional chaining)?

Reply:
ah! yes of course. not sure how that got lost